### ISSUE TITLE
[Feature Request] MS Teams

### ISSUE THREAD

**Issue by phrichuk** [AutoDarkMode/Windows-Auto-Night-Mode#162]
Great utility! Good to have MS Office covered now. Could MS Teams App be considered and 'honorary member' of the Office suite in a future version? Like Office, it has decent default and dark themes as manual menu options, but as yet no elegant, automated way to switch between them... Apologies if this has already been suggested; I could not see it in the list. Thanks.

**Comment by mateuszdrab** [comment]
Office seems to have auto theme switching now. Teams however, still doesn't and it's been two years according to:
https://teamsfeedback.uservoice.com/forums/427395-internal-microsoft/suggestions/35729917-automatically-switch-to-dark-theme-when-in-mac-or
I too would benefit from this being a feature until it's implemented officially (probably somewhere in 2055) 😂

**Comment by emberaholdon** [comment]
> Office seems to have auto theme switching now. Teams however, still doesn't and it's been two years according to:
> https://teamsfeedback.uservoice.com/forums/427395-internal-microsoft/suggestions/35729917-automatically-switch-to-dark-theme-when-in-mac-or
> I too would benefit from this being a feature until it's implemented officially (probably somewhere in 2055) 😂
> I'm probably going to have to code this myself, providing there is a way to make teams switch theme externally.

Correct me if I’m wrong but Office only changes themes when you restart the apps. When running Outlook all day it stays black/white regardless of theme changjng of Windows.

**Comment by mateuszdrab** [comment]
@emberaholdon for me it changes theme but Outlook forgets to update the reading pane so it sometimes stays white when the rest of the UI is dark and it needs a restart to take effect.

**Comment by emberaholdon** [comment]
> @emberaholdon for me it changes theme but Outlook forgets to update the reading pane so it sometimes stays white when the rest of the UI is dark and it needs a restart to take effect.

I have never experienced any Office automatic theme change, with any version of Windows or Office. They only change theme when I start a new instance of the program.

**Comment by mateuszdrab** [comment]
Have you tried this?

[cid:ee0d990c-e548-4949-b3ec-6444d6983b08]
________________________________

**Comment by mateuszdrab** [comment]
Have you tried this?

![image](https://user-images.githubusercontent.com/26556541/99550475-0d147600-29bb-11eb-9045-b825b4d99c3f.png)

**Comment by emberaholdon** [comment]
> Have you tried this?

Are You using an insider build?

![image](https://user-images.githubusercontent.com/67831954/99550699-4a790380-29bb-11eb-9b20-ae2a6db71955.png)

**Comment by emberaholdon** [comment]
> > Have you tried this?
> 
> Are You using an insider build?
> 
> ![image](https://user-images.githubusercontent.com/67831954/99550699-4a790380-29bb-11eb-9b20-ae2a6db71955.png)

I managed to find out: the "current channel (preview)" has this feature, the "current channel" has not - even though the build number is much higher... Microsofts versioning system is a freacking mess.

**Comment by mateuszdrab** [comment]
You're right. I think I was on an internal build - I don't have that option now. However, Teams really needs attention. Using the web browser app for now is a bit of an overkill. VScode could also benefit from this amazing app.
When I get some time, I may have to look into adding the feature in myself.

**Comment by chonty** [comment]
@mateuszdrab VS Code already has extensionless system-defined theming
![image](https://user-images.githubusercontent.com/20431435/118198517-6a820f00-b494-11eb-9aaa-9bf19e3a66cb.png)

**Comment by mateuszdrab** [comment]
@chonty that's cool. I didn't know that.
That leaves Teams as the only thing that doesn't support auto dark mode.

**Comment by eveneveneven** [comment]
Any news on this? Anyone know if it's possible to change the Teams Theme externally?
I'd love to try to implement it, but I'm not sure where to begin with changing Teams settings from an external program.

**Comment by emberaholdon** [comment]
> Any news on this? Anyone know if it's possible to change the Teams Theme externally?
> I'd love to try to implement it, but I'm not sure where to begin with changing Teams settings from an external program.

Until this is solved, I would suggest that You try the PWA (web app) version of Teams. You can install it by going to teams.microsoft.com in Edge. If you use the Dark Reader extension in Edge, it can automatically change all websites (including PWAs) to dark mode.

**Comment by mateuszdrab** [comment]
> Any news on this? Anyone know if it's possible to change the Teams Theme externally? I'd love to try to implement it, but I'm not sure where to begin with changing Teams settings from an external program.

I'm pretty sure that Teams stores the theme in the desktop-config.json file in AppData\Roaming\Microsoft\Teams

There is a property called theme that has value of either defaultV2 or darkV2.

Unfortunately changing this file doesn't result in anything since it gets overwritten by teams. I'm not sure how to trigger teams to reload without restarting it.

**Comment by Enf0** [comment]
Hi,

Just wanted to let you know that the upcoming "Teams 2.0" has an option to follow the system theme, unlike the current version of Teams.

The new version is already available for personal use; no word yet from Microsoft from when it will be available for school/work accounts, unfortunately.

![image](https://user-images.githubusercontent.com/825808/146635352-22b20f2f-2506-430e-bddb-9cf6afa43e01.png)

**Comment by carlosvsilva** [comment]
This never made it to Teams (work/school accounts) did it ?

**Comment by mateuszdrab** [comment]
> This never made it to Teams (work/school accounts) did it ?

Nope :(

**Comment by kevin-ctera** [comment]
Does anyone know why is this still missing from work/school teams?

**Comment by emberaholdon** [comment]
> Does anyone know why is this still missing from work/school teams?

The new preview version includes this option.

https://techcommunity.microsoft.com/t5/microsoft-teams-blog/introducing-the-new-microsoft-teams-now-in-preview/ba-p/3774406
https://learn.microsoft.com/en-gb/microsoftteams/public-preview-doc-updates

**Comment by kevin-ctera** [comment]
> > Does anyone know why is this still missing from work/school teams?
> 
> The new preview version includes this option.
> 
> https://techcommunity.microsoft.com/t5/microsoft-teams-blog/introducing-the-new-microsoft-teams-now-in-preview/ba-p/3774406 https://learn.microsoft.com/en-gb/microsoftteams/public-preview-doc-updates

Downloaded from the following [link](https://statics.teams.cdn.office.net/production-windows-x64/23113.200.2039.1130/MSTeams-x64.msix) which was 2.1 preview located [here](https://github.com/ItzLevvie/MicrosoftTeams-msinternal/blob/master/defconfig2) and it works great!

![image](https://user-images.githubusercontent.com/119529412/235232213-29ddf178-f6c7-4b17-a6ca-c1b93dcc4505.png)

Thanks, @emberaholdon !!!

**Comment by bcutter** [comment]
Too bad this seems to be out of control auf ADM.

Hope to get a ping here once the series 2 version of Teams (for work/business) has been released, finally being able to have the automatic theme switch.

**Comment by emberaholdon** [comment]
> Too bad this seems to be out of control auf ADM.
> 
> Hope to get a ping here once the series 2 version of Teams (for work/business) has been released, finally being able to have the automatic theme switch.

The new Teams client already has a dynamic theme option.

**Comment by emberaholdon** [comment]
> Latest Teams for work is v1.6 - which does not have this option yet.

You have to turn on the “new” Teams (switch in the upper left corner) and then you get this version:

![IMG_5679](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/67831954/ae6ce5be-f7e7-4409-8e46-48415066500d)
![IMG_5680](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/67831954/2751d8f6-3ac6-43ec-b31a-e4e4c0d7aefc)

**Comment by bcutter** [comment]
I tend to not getting into a Teams discussion here. My last words on this: Not available, likely corporate setting (as I'm as repeatedly said talking about Teams **for work**).

**Comment by emberaholdon** [comment]
> I tend to not getting into a Teams discussion here. My last words on this: Not available, likely corporate setting (as I'm as repeatedly said talking about Teams **for work**).

This is Teams for work. Your company policy likely just prevents you from upgrading to the new version.

**Comment by ChenYiLins** [comment]
Is this still a problem now? 🤔

**Comment by mateuszdrab** [comment]
> Is this still a problem now? 🤔

I don't think so, new Teams comes with system managed theme support by default.

**Comment by emberaholdon** [comment]
Indeed, Office 365 apps and Teams are changing themes dynamically following the system setting. No need to restart the apps or anything.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/issues/162#issuecomment-1689656016

@emberaholdon, the PWA, installable via Chromium?
